astropy/coordinates/angle_formats.py

```diff
@@ -412,13 +412,22 @@ def degrees_to_dms(d):
 def dms_to_degrees(d, m, s=None):
     """
     Convert degrees, arcminute, arcsecond to a float degrees value.
-    """
-
+     """
     _check_minute_range(m)
     _check_second_range(s)
+    
+    # if d is a scaler
+    d = np.atleast_1d(d)
+    m = np.atleast_1d(m)
+    s = np.atleast_1d(s)
+    dmss = np.array([d,m,s]).T
+
+    for dms in dmss:
+        # looking for first non zero
+        for i in dms:
+            if i!=0:
+                sign = np.copysign(1,i); break;
```

Comment:
This is implementation only works for scalars

Let's look at the following example

```python
>>> d = np.array([  0.,  45., -45.])
>>> m = np.array([  0.,  30., -30.])
>>> s = np.array([ 0.,  0., -0.])
>>> for dms in np.array([d,m,s]).T:
...     print(dms)
...     for i in dms:
...         print(i)
...         if i != 0:
...             sign = np.copysign(1, i)
...             break
>>> print("sign =", sign)
[0. 0. 0.]
0.0
0.0
0.0
[45. 30.  0.]
45.0
[-45. -30.  -0.]
-45.0
sign = -1.0
```
So the only sign that gets recorded is the last one.
Also, in the case of d=m=s=0, this actually never even assigns a value to 'sign'!

```python
>>> for dms in dmss:
...     for i in dms:
...         if i != 0:
...             sign = np.copysign(1, i)
...             break
...     print("current sign =", sign)
NameError: name 'sign' is not defined
```